app/mailers/user_mailer.rb

```diff
@@ -0,0 +1,10 @@
+class UserMailer < ApplicationMailer
+
+  def confirmation(user)
+    @user = user
+
+    mail(to: @user.email, subject: 'Account Confimration')
+
+    @user.update(confirmation_token_sent_at: Time.current) 
```

Comment:
this should be set when confirmation_token is genereated

Reply:
secondly when the account is confirmed the confirmation_token should be set to nil